### PR TITLE
fix(DTFS2-6147): vulnerabilities test coverage 

### PR DESCRIPTION
**Pull request by Zainzzkk**
## Introduction
Various vulnerabilities test coverage has been missing.

## Resolution
* Added cypress test for NoSQL injection input
* Added api-test for users.api-test for login with NoSQL injection

### CODE REVIEW

File: e2e-tests/portal/cypress/e2e/journeys/admin/create-user/create-a-user.spec.js [hunk 1]
```diff
@@ -25,6 +25,16 @@ context('Admin user creates a new user', () => {
     roles: ['maker'],
   };
 
+  const userWithInjection = {
+    username: '"$gt": ""',
+    email: '"$gt": ""',
```
Comment (abhi-markan) [inline e2e-tests/portal/cypress/e2e/journeys/admin/create-user/create-a-user.spec.js:30]: Can you please change this to `{ "$gt": "" }`

**Review by abhi-markan — CHANGES_REQUESTED** [review]

File: e2e-tests/portal/cypress/e2e/journeys/admin/create-user/create-a-user.spec.js (hunk 1 above)
Comment (Zainzzkk) [reply e2e-tests/portal/cypress/e2e/journeys/admin/create-user/create-a-user.spec.js]: changed

**Review by IgnatG — DISMISSED** [review]

File: e2e-tests/portal/cypress/e2e/journeys/admin/create-user/create-a-user.spec.js [hunk 2]
```diff
@@ -131,5 +141,50 @@ context('Admin user creates a new user', () => {
...
+    });
+
+    /**
+     * to check that user has not been created
+     * gets list of users from portal-api
+     * finds one with email { "$gt": "" }
+     * should be undefined
+     */
+    cy.listAllUsers().then((usersInDb) => {
+      const injectedUser = usersInDb.find((user) => user.email === userWithInjection.email);
```
Comment (abhi-markan) [inline e2e-tests/portal/cypress/e2e/journeys/admin/create-user/create-a-user.spec.js:183]: This needs to be updated to following as the user with email will be created with email as `email: {}`

```javascript
const injectedUser = usersInDb.find((user) => user.email === {});
```

File: portal-api/api-tests/v1/users/users.api-test.js [hunk 3]
```diff
@@ -246,4 +246,73 @@ describe('a user', () => {
...
+        await as().post(injectedUser).to('/v1/users');
+
+        const { status, body } = await as().post({ username, password }).to('/v1/login');
+
+        expect(status).toEqual(401);
+        expect(body).toEqual(expectedBody);
+      });
+
+      describe('when username is "{ "$lt": "" }"', () => {
+        it('should return a user cannot be found message', async () => {
```
Comment (abhi-markan) [inline portal-api/api-tests/v1/users/users.api-test.js:300]: Can we create an assertion where no injected user is inserted and you simply login directly with following payload please?

```javascript
{
    "username": "{$or: [{role: { $ne: '' }}]}",
    "password": "AbC!2345"
}
```

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Two minor comments to address please.

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Just a minor one please.

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**